crates/net/eth-wire/src/types/message.rs

```diff
@@ -214,6 +214,23 @@ impl EthMessage {
     }
 }
 
+trait EncodableExt {
+    fn encode_max(&self, size: usize) -> alloy_rlp::Result<Vec<u8>, String>;
+}
```

Comment:
we want two functions here,

one that returns an error and one that truncates: put as many encoded items in the buffer as possible